python/plugins/processing/algs/qgis/VectorLayerScatterplot.py

```diff
@@ -166,7 +182,55 @@ def processAlgorithm(self, parameters, context, feedback):
 
         output = self.parameterAsFileOutput(parameters, self.OUTPUT, context)
 
-        values = vector.values(source, xfieldname, yfieldname)
+        hoverexpression = self.parameterAsExpression(
+            parameters, self.HOVERTEXT, context
+        )
+        if hoverexpression.strip() != "":
+            exp_context = QgsExpressionContext()
+            vlayer = QgsProcessingUtils.mapLayerFromString(
+                parameters[self.INPUT], context
+            )
+            exp_context.appendScopes(
+                QgsExpressionContextUtils.globalProjectLayerScopes(vlayer)
+            )
```

Comment:
This will do all of this, in a safer way and in a way that works correctly with the wider context when the algorithm is run as part of a model:
```suggestion
            exp_context = self.createExpressionContext(parameters, context, source)
```